src/ResponseGenerator.py

```diff
@@ -0,0 +1,50 @@
+from transformers import AutoModelForCausalLM, AutoTokenizer
+import torch
+
+class ResponseGenerator():
+
+    def __init__(self, model_name, chat_history_ids="default",bot_input_ids="default"):
+        self.processor = AutoTokenizer.from_pretrained(model_name)
+        self.model = AutoModelForCausalLM.from_pretrained(model_name)
+        self.chat_history_ids=chat_history_ids
+        self.bot_input_ids=bot_input_ids
+
+    def encoder(self, step):
```

Comment:
I think the encoder should take in the user query so I can just call `response_generator.encoder(user_query)`